include/envoy/http/codec.h

```diff
@@ -148,11 +148,28 @@ class ConnectionCallbacks {
 };
 
 /**
- * A list of options that can be specified when creating a codec.
+ * http/2 settings
  */
-class CodecOptions {
-public:
-  static const uint64_t NoCompression = 0x1;
+struct Http2Settings {
+  /**
+   * A list of options that can be specified when creating a codec.
+   */
+  struct CodecOptions {
+    static const uint64_t DisableDynamicHPACKTable = 0x1;
+  };
+
+  Http2Settings()
+      : codec_options_(0), max_concurrent_streams_(DEFAULT_MAX_CONCURRENT_STREAMS),
+        initial_window_size_(DEFAULT_INITIAL_WINDOW_SIZE) {}
+
+  uint64_t codec_options_;
+  uint32_t max_concurrent_streams_;
+  uint32_t initial_window_size_;
+
+  static const uint32_t DEFAULT_MAX_CONCURRENT_STREAMS = 1024;
```

Comment:
We don't really have consistent style here unfortunately. I would probably s/DisableDynamicHPACKTable/DISABLE_DYNAMIC_HPACK_TABLE